projects/angular/components/ui-grid/src/ui-grid.component.ts

```diff
@@ -368,6 +369,23 @@ export class UiGridComponent<T extends IGridDataEntry> extends ResizableGrid<T>
      *
      */
     public isAnyFilterDefined$ = new BehaviorSubject<boolean>(false);
+    /**
+     * Array of columns with groupable directive, if any column is present a groupBy dropdown should appear
+     *
+     */
+    public groupableColumns = [] as UiGridColumnDirective<T>[];
+    /**
+     * Array of groups and its state
+     *
+     */
+    public groups: { propertyValue: UiGridColumnDirective<T>['property'], message: any, collapsed?: boolean }[] = [];
+
+    /**
+     * group header message
+     *
+     */
+    public groupMessage: any;
```

Comment:
shouldn't this be a string?
```suggestion
    public groupMessage: string;
```

Reply:
Can't columns's property Value can be of type number, enum, etc other than string?

Reply:
The value, sure... But the group`Messsage` should be a string

Reply:
makes sense